source/tokens.rs

```diff
@@ -824,6 +831,81 @@ pub struct Span<'a> {
     slice: Input<'a>
 }
 
+impl<'a> InputTakeAtPosition for Span<'a> {
+  type Item = InputElement;
+
+  fn split_at_position<P>(&self, predicate: P) -> IResult<Self, Self, u32>
+  where
+    P: Fn(Self::Item) -> bool,
+  {
+    match (0..self.slice.len()).find(|b| predicate(self.slice[*b])) {
+      Some(i) => Ok((Span {
+          offset: self.offset,
+          line: self.line,
+          column: self.column,
+          slice: &self.slice[i..],
+      }, Span {
+          offset: self.offset,
+          line: self.line,
+          column: self.column,
+          slice: &self.slice[..i],
+      })),
+      None => Err(Err::Incomplete(Needed::Size(1))),
+    }
+  }
+
+  fn split_at_position1<P>(&self, predicate: P, e: ErrorKind<u32>) -> IResult<Self, Self, u32>
+  where
+    P: Fn(Self::Item) -> bool,
+  {
+    match (0..self.slice.len()).find(|b| predicate(self.slice[*b])) {
+      Some(0) => Err(Err::Error(Context::Code(*self, e))),
+      Some(i) => Ok((Span {
```

Comment:
@Hywan So I found one of the many unknown reasons why the tests are failing.

One problem is that the new `Span` is not passing the correct `offset`, `line`, `column` values - after the split. I am looking for examples where, after the `slice` is converted into `utf8`, the values are computed using a function or something. There are tabs and newslines in the input, it should return the correct `offset` and `line` after converting it into `utf8`. Do you know any such example?

For instance:

```rs
    fn case_first_with_whitespace() {
        named!(hello<Span, Span>, tag!(b"hello"));
        named!(test1<Span, Span>, first!(tag!(b"hello")));
        named!(test2<Span, Span>, first!(hello));

        let input  = Span::new(b"  \nhello\t\r");
        let output = Ok((Span::new_at(b"\t\r", 8, 2, 6), Span::new_at(b"hello", 3, 2, 1)));

        assert_eq!(test1(input), output);
        assert_eq!(test2(input), output);
    }
```

Fails with out output:

```
Diff < left / right > :                                                                                                                                                     
 Ok(                                                                                                                                                                        
     (                                                                                                                                                                      
         Span {                                                                                                                                                             
<            offset: 0,                                                                                                                                                     
<            line: 1,                                                                                                                                                       
<            column: 1,                                                                                                                                                     
>            offset: 8,                                                                                                                                                     
>            line: 2,                                                                                                                                                       
>            column: 6,                                                                                                                                                     
             slice: [                                                                                                                                                       
                 9,                                                                                                                                                         
                 13                                                                                                                                                         
             ]                                                                                                                                                              
         },                                                                                                                                                                 
         Span {                                                                                                                                                             
<            offset: 0,                                                                                                                                                     
<            line: 1,                                                                                                                                                       
>            offset: 3,                                                                                                                                                     
>            line: 2,                                                                                                                                                       
             column: 1,                                                                                                                                                     
             slice: [                                                                                                                                                       
                 104,                                                                                                                                                       
                 101,                                                                                                                                                       
                 108,                                                                                                                                                       
                 108,                                                                                                                                                       
                 111                                                                                                                                                        
             ]                                                                                                                                                              
         }                                                                                                                                                                  
     )                                                                                                                                                                      
 )
```

Reply:
When computing the new `Span`, you should update the `offset` to include `i` (from `find`). Same for `line` and `column`.

You might want to check https://github.com/tagua-vm/parser/blob/077aa7ae3153fa8ae1dbb5b19076ca4c3cb86951/source/tokens.rs#L1180-L1222 to see an example of that logic. One way to do that would be (I imagine):

```rust
Ok((self.slice(i..), self.slice(..i))
```

Because `slice` will return a `Span` with appropriated values set.